### DISCUSSION TITLE
Clean build yields: warn  - You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env

### DISCUSSION THREAD

**Discussion by tippfelher** [vercel/next.js#46638] [Help] [answered]
I created a brand new nextjs project using `npx create-next-app@latest .`.

Without changing anything I invoked `yarn next build`. It outputs:

> warn  - You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env

1) Reading through some github comments and the contents under the link above, my understanding is that `next build` should always use `production`. So my first question is: Why does the build script even complain here?

2) Since the warning did not even mention WHAT NODE_ENV is set I added a debug output to `next.config.js`: `console.log('DBG_NODE_ENV', process.env.NODE_ENV);` The output was `DBG_NODE_ENV development` during build process. While  indeed `development` != `production` it's still a valid env value?

3) Trying to set the NODE_ENV in the package.json build command (for example: `"build": "cross-env NODE_ENV=production next build",`) also didn't change anything.

I am completely lost. How do I get rid of the warn message and have nextjs actually build with NODE_ENV=production?

**Comment by wwwdepot** [comment]
Build command is always set to production.

**Reply by tippfelher** [reply]
In theory, but if you would have read my post, this was not the case.

**Comment by tippfelher** [comment, marked as answer]
Solution: There was a global NODE_ENV environment variable in my system. (Windows 10) I removed it and now `next build` uses `production` instead of `development`.
